inference-engine/src/vpu/common/src/ngraph/transformations/dynamic_to_static_shape.cpp

```diff
@@ -119,6 +133,11 @@ DynamicToStaticShape::DynamicToStaticShape(const Transformations& specificTransf
 
 bool DynamicToStaticShape::run_on_function(std::shared_ptr<ngraph::Function> function) {
     bool function_changed = false;
+
+    // Ensure that existing DSRs in function propagate upper-bound shapes, not dynamism.
+    // Basically this is possible in test cases, when the function is initially configured with DSR as inputs.
+    function_changed |= propagateUpperBoundFromExistingDSR(function);
```

Comment:
Could you please explain a bit more why we need this? I got the part that we have a test with function containing DSRs. But why they need to propagate upper-bound sizes?

Reply:
I did this in order to keep the dynamic state of the function up to this point, and not to emulate it in tests. If the DSR propagates dynamism (not upper-bound), then that dynamism will be maintained with every function revalidation. 

Now the function created in the tests gets to this point with dynamism on the output edges of the operations, but the DSR mode is set to `INFER_DYNAMIC_SHAPE`. We need to switch it to the `INFER_UPPER_BOUND_SHAPE` mode in order to estimate the shapes of tensors.

Thus, now we can make a test with dynamic input (that is, with DSR at the input) and not hardcode the output type via `set_output_type(i, elemType, dynamicPartialShape)`, and also not call DTS transformations explicitly in the test body, but wait for the plugin to do it itself.

Reply:
Discussed offline.